crates/bevy_ecs/src/system/commands/mod.rs

```diff
@@ -685,9 +787,49 @@ where
         if let Some(mut entity) = world.get_entity_mut(self.entity) {
             entity.insert_bundle(self.bundle);
         } else {
-            panic!("Could not insert a bundle (of type `{}`) for entity {:?} because it doesn't exist in this World.\n\
-                    If this command was added to a newly spawned entity, ensure that you have not despawned that entity within the same stage.\n\
-                    This may have occurred due to system order ambiguity, or if the spawning system has multiple command buffers", std::any::type_name::<T>(), self.entity);
+            panic_fmt::<T>("insert", "bundle", "for", self.entity);
+        }
+    }
+}
+
+pub struct MoveBundle<T> {
+    pub source: Entity,
+    pub target: Entity,
+    _bundle: PhantomData<T>,
+}
+
+impl<T> MoveBundle<T> {
+    /// Creates a new [`MoveBundle`] with given source and target IDs.
+    pub fn new(source: Entity, target: Entity) -> Self {
+        Self {
+            source,
+            target,
+            _bundle: PhantomData,
+        }
+    }
+}
+
+impl<T> Command for MoveBundle<T>
+where
+    T: Bundle + 'static,
+{
+    fn write(self, world: &mut World) {
+        let bundle_opt = if let Some(mut source) = world.get_entity_mut(self.source) {
+            source.remove_bundle::<T>()
+        } else {
+            panic_fmt::<T>("move", "bundle", "from", self.source);
+        };
+
+        let bundle = if let Some(some) = bundle_opt {
+            some
+        } else {
+            return;
```

Comment:
should it panic in this case?

Reply:
If we're being consistent, yes.

I dislike this behavior, but that's for a different PR.